imap_processing/hit/l0/decom_hit.py

```diff
@@ -0,0 +1,464 @@
+    start_index : int
+        Starting index to find the first packet in a
+        science frame from an array of sequence flags.
+
+    Returns
+    -------
+    flag_index : int
+        Index of starting packet for next science frame.
+    """
+    for flag_index, flag in enumerate(seq_flgs[start_index:]):
+        if flag == 1:
+            # return starting index of next science frame
+            return flag_index + start_index
+    return None
+
+
+def is_valid_science_frame(seq_flgs: np.ndarray, src_seq_ctrs: np.ndarray) -> bool:
+    """
+    Check for valid science frame.
+
+    Each science data packet has a sequence grouping flag that can equal
+    0, 1, or 2. These flags are used to group 20 packets into science
+    frames. This function checks if the sequence flags for a set of 20
+    data packets have values that form a complete science frame.
+    The first packet should have a sequence flag equal to 1. The middle
+    18 packets should have a sequence flag equal to 0 and the final packet
+    should have a sequence flag equal to 2.
+
+    Valid science frame sequence flags
+    [1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2]
+
+    Additionally, packets have a sequence counter. This function also
+    checks if the counters are in sequential order.
+
+    Both conditions need to be met for a science frame to be considered
+    valid.
+
+    Parameters
+    ----------
+    seq_flgs : numpy.ndarray
+        Array of sequence flags for a set of 20 packets.
+    src_seq_ctrs : numpy.ndarray
+        Array of sequence counters for a set of 20 packets.
+
+    Returns
+    -------
+    boolean : bool
+        Boolean for whether the science frame is valid.
+    """
+    # Check sequence grouping flags
+    if not (seq_flgs[0] == 1 and all(seq_flgs[1:19] == 0) and seq_flgs[19] == 2):
+        # TODO log issue
+        print(f"Invalid seq_flgs found: {seq_flgs}")
+        return False
+
+    # Check that sequence counters are sequential
+    if not np.all(np.diff(src_seq_ctrs) == 1):
+        # TODO log issue
+        print(f"Non-sequential src_seq_ctr found: {src_seq_ctrs}")
+        return False
+
+    return True
+
+
+def update_ccsds_header_data(sci_dataset: xr.Dataset) -> xr.Dataset:
+    """
+    Update dimensions of CCSDS header fields.
+
+    The CCSDS header fields contain 1D arrays with
+    values from all the packets in the file. This
+    function updates the dimension for these fields
+    to use sc_tick instead of epoch. sc_tick is the
+    time the packet was created.
+
+    Parameters
+    ----------
+    sci_dataset : xr.Dataset
+        Xarray dataset containing HIT science packets
+        from a CCSDS file.
+
+    Returns
+    -------
+    sci_dataset : xr.Dataset
+        Updated xarray dataset.
+    """
+    # sc_tick contains spacecraft time per packet
+    sci_dataset.coords["sc_tick"] = sci_dataset["sc_tick"]
+    sci_dataset = sci_dataset.swap_dims({"epoch": "sc_tick"})
```

Comment:
I am not sure if you need this function. I believe `packet_file_to_datasets` already creates `epoch` using `sc_tick`.

Reply:
I replace the epoch dimension later to only contain one time per science frame rather than per packet since that is what's needed for the science data. However, I still need the per packet sc_tick times for the ccsds header fields since those are not being grouped by science frame. Essentially, I need two time dimensions. One per packet for the ccsds header fields and one per science frame for the science data. I can explain this better in the docstring